browser/features.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/features.h"
+
+#include "build/build_config.h"
+
+static_assert(BUILDFLAG(IS_MAC), "Currently for macOS only");
```

Comment:
Thank you for the inputs. I have a different opinion about this particular point and will try to explain my reasoning below. Is it OK with you if I merge the PR as-is?

My preference is for code to do as little as possible to achieve the current desired functionality. This way, when I see code, I know that it is used, I know that I need to change it, and I know that changing it will have a testable effect. When code is generic, and covers scenarios that are actually not used in practice, then I'm left wondering "is this actually used?", "do my changes cover all scenarios?", "am I really testing all scenarios?"

In this specific example, `features` is currently only used on macOS. And the code makes this clear, for example via the `static_assert` above. But imagine if `features` were available on all platforms. Even though it is entirely unused there, future Windows developers touching the code in any way would have to first realize that, actually, the code currently has no effect on Windows, that they don't need to test there unless they make changes, etc.

I understand your point about the cost of unwrapping the `is_mac` guards. But it seems to me that the costs of doing it now, versus doing it in the future, are the same. So it seems to me that it is better not to incur that cost now, especially in light of the argument made in the previous paragraph. Is that sensible?

Reply:
> Is it OK with you if I merge the PR as-is?

Yes.

> My preference is for code to do as little as possible to achieve the current desired functionality.

It’s perfectly fine if you name the target `updater_features` or `omaha_features` or `mac_features`. However, by introducing a generic `//brave/browser:features` target, you set certain expectations among Chromium developers. Having the `//brave/browser:features` target available only on a single platform feels counter-intuitive and may cause confusion. I hope this concern is clear.

Reply:
Yup, that makes sense. Do you think I should rename it to `updater_features`?

Reply:
> Yup, that makes sense. Do you think I should rename it to `updater_features`?

Any non-generic name will be fine. `updater_features` sgtm.

Reply:
I figured that `mac_features` is actually better and renamed it to this now in ef2df6c1a026ce9b1ffd796d0e64f0775dd3e1b1.